x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields.tsx

```diff
@@ -76,80 +81,126 @@ const columns: Array<EuiBasicTableColumn<HighlightedFieldsTableRow>> = [
       values: string[] | null | undefined;
       scopeId: string;
       isPreview: boolean;
+      showCellActions: boolean;
     }) => (
-      <CellActions field={description.field} value={description.values}>
-        <HighlightedFieldsCell
-          values={description.values}
-          field={description.field}
-          originalField={description.originalField}
-        />
-      </CellActions>
+      <>
+        {description.showCellActions ? (
+          <CellActions field={description.field} value={description.values}>
+            <HighlightedFieldsCell
+              values={description.values}
+              field={description.field}
+              originalField={description.originalField}
+              scopeId={description.scopeId}
+            />
+          </CellActions>
+        ) : (
+          <HighlightedFieldsCell
+            values={description.values}
+            field={description.field}
+            originalField={description.originalField}
+          />
+        )}
+      </>
     ),
   },
 ];
 
+export interface HighlightedFieldsProps {
+  /**
+   * An array of field objects with category and value
+   */
+  dataFormattedForFieldBrowser: TimelineEventsDetailsItem[];
+  /**
+   * User defined fields to highlight (defined on the rule)
+   */
+  investigationFields: string[];
+  /**
+   * Boolean to indicate whether flyout is opened in rule preview
+   */
+  isPreview: boolean;
+  /**
+   * Maintain backwards compatibility // TODO remove when possible
+   * Only needed if alerts page flyout (which uses CellActions), NOT in the AI for SOC alert summary flyout.
+   */
+  scopeId?: string;
```

Comment:
same here, instead of making it optional with a call out, how about adding a separate prop called `disablePreview`? there will be some prop drilling, but it's more comprehendible

Reply:
Yup I actually update that change locally. Will push the commit soon, finishing unit tests!

Reply:
So I actually approached this differently: I kept the `scopeId` optional (because we're optionally passing it to `CellActions`, but I do have a `showPreview` in the `HighlightedFieldsCell` component